docs/refmans/gui/viselements/generic/button.md_template

```diff
@@ -2,44 +2,105 @@ A control that can trigger a function when pressed.
 
 # Usage
 
-## Simple button
+## Simple button {data-source="gui:doc/examples/controls/button_simple.py"}
 
-The button label, which is the button control's default property, is simply displayed as the button
-text.
+The label of the button, defined by the default property [*label*](#p-label) of the button control,
+is displayed as the text on the button.
 
+Below is an example of how to define a button:
 !!! taipy-element
     default=Button Label
 
+Here is how the button is displayed:
 <figure>
     <img src="../button-simple-d.png" class="visible-dark" />
     <img src="../button-simple-l.png" class="visible-light"/>
     <figcaption>A simple button</figcaption>
 </figure>
 
-## Specific action callback
+## Specific action callback {data-source="gui:doc/examples/controls/button_action.py"}
 
-Button can specify a callback function to be invoked when the button is pressed.
+A button can specify a callback function to be invoked when it is pressed. By default, the global
+function *on_action()* is triggered by the `on_action` callback if it is defined, as described in
+the [section on callbacks](../../../../userman/gui/callbacks.md).
 
-If you want some function called *button_pressed()* to be invoked when the user presses a button,
-you can define this function as follows:
-```py
+To define a custom callback function for a button press, you can set the [*on_action*](#p-on_action)
+property to reference the specific callback function you want to invoke.
+
+Here is an example of defining such a callback function, named *button_pressed()*:
+```python
 def button_pressed(state):
-  # React to the button press action
+    # React to the button press action
 ```
 
-Then refer to this function in the definition of the control, as the value of the button's
-[*on_action*](#p-on_action) property:
-
+You can then assign this function to the button's [*on_action*](#p-on_action) property:
 !!! taipy-element
     default=Button Label
     on_action:f=button_pressed
 
+When the user presses the button, the callback function *button_pressed()* is triggered.
+
+The appearance of the button remains the same, even with a callback function bound to it.
+
+## Action as a lambda function {data-source="gui:doc/examples/controls/button_lambda.py"}
+
+A callback function can be defined as a lambda function to handle button actions.
+
+Below is an example of defining a button using a lambda function for the [*on_action*](#p-on_action)
+property:
+!!! taipy-element
+    default=Button Label
+    id=button1
+    on_action:f=lambda s, i, p: s.assign("text", f"button '{i}': '{p.get('action')}'.")
```

Comment:
More descriptive parameter names? 
State, id, payload

Reply:
That would make the line a bit too long (in the Builder construct) in my opinion...